File: scavenger-hunt-app/src/models/Quest.ts
import { Waypoint } from './Waypoint';

export interface Quest {
  id: string;
  name: string;
  description: string;
  difficulty: 'easy' | 'medium' | 'hard';
  estimatedTime: number; // w minutach
  waypoints: Waypoint[];
  isActive: boolean;
  isCompleted: boolean;
  createdAt: Date;
} 

File: ScavengerHunt.Api/Models/Quest.cs
namespace ScavengerHunt.Api.Models;

public class Quest
{
    public string Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public string Difficulty { get; set; } // "easy", "medium", "hard"
    public int EstimatedTime { get; set; } // in minutes
    public List<Waypoint> Waypoints { get; set; }
    public bool IsActive { get; set; }
    public bool IsCompleted { get; set; }
    public DateTime CreatedAt { get; set; }
}